plugins/accessibility/index.js

```diff
@@ -32,76 +57,187 @@ var AUDIT_FILE = path.join(__dirname, '../../node_modules/accessibility-develope
  */
 function teardown(config) {
 
+  var audits = [];
+
   if (config.chromeA11YDevTools) {
+    audits.push(runChromeDevTools(config));
+  }
+  // check for Tenon config and an actual API key, not the placeholder
+  if (config.tenonIO && /[A-Za-z][0-9]/.test(config.tenonIO.options.key)) {
+    audits.push(runTenonIO(config));
+  }
+  return q.all(audits).then(function() {
+    return outputResults();
+  });
+}
 
-    var data = fs.readFileSync(AUDIT_FILE, 'utf-8');
-    data = data + ' return axs.Audit.run();';
+var testOut = {failedCount: 0, specResults: []};
+var entities = new Entities();
 
-    var testOut = {failedCount: 0, specResults: []},
-        elementPromises = [];
+/**
+ * Audits page source against the Tenon API, if configured. Requires an API key:
+ * more information about licensing and configuration available at
+ * http://tenon.io/documentation/overview.php.
+ *
+ * @param {Object} config The configuration file for the accessibility plugin
+ * @return {q.Promise} A promise which resolves to the results of any passed or
+ *    failed tests
+ * @private
+ */
+function runTenonIO(config) {
+  var options = config.tenonIO.options;
+
+  return browser.driver.getPageSource().then(function(source) {
+    // setup response as a deferred promise
+    var deferred = q.defer();
+    request.post({
+      url: TENON_URL,
+      form: {
+        key: options.key,
+        src: source,
+        level: options.level,
+        filter: options.filter,
+        certainty: options.certainty,
+        store: options.store
+      }
+    },
+    function(err, httpResponse, body) {
+      if (err) return resolve.reject(new Error(err));
+      else return deferred.resolve(JSON.parse(body));
+    });
 
-    return browser.executeScript_(data, 'a11y developer tool rules').then(function(results) {
+    return deferred.promise.then(function(response) {
+      return processTenonResults(response);
+    });
+  });
 
-      var audit = results.map(function(result) {
-        var DOMElements = result.elements;
-        if (DOMElements !== undefined) {
+  function processTenonResults(response) {
+    var numResults = response.resultSet.length;
 
-          DOMElements.forEach(function(elem) {
-            // get elements from WebDriver, add to promises array
-            elementPromises.push(
-              elem.getOuterHtml().then(function(text) {
-                return {
-                  code: result.rule.code,
-                  list: text
-                };
-              })
-            );
-          });
-          result.elementCount = DOMElements.length;
-        }
-        return result;
+    testOut.failedCount = numResults;
+
+    var testHeader = 'Tenon.io - ';
+
+    if(numResults === 0) {
+      return testOut.specResults.push({
+        description: testHeader + 'All tests passed!',
+        assertions: [{
+          passed: true,
+          errorMsg: ''
+        }],
+        duration: 1
```

Comment:
In the Tenon response we get some info, can we make use of either of these properties?

```json
 responseExecTime: '0.69',
 responseTime: '2015-02-27T19:27:59.060Z',
```